app/assets/javascripts/components/AccountMenu/Authentication.tsx

```diff
@@ -0,0 +1,382 @@
+import { confirmDialog } from '@Services/alertService';
+import {
+  STRING_ACCOUNT_MENU_UNCHECK_MERGE,
+  STRING_GENERATING_LOGIN_KEYS,
+  STRING_GENERATING_REGISTER_KEYS,
+  STRING_NON_MATCHING_PASSWORDS
+} from '@/strings';
+import { JSXInternal } from '@node_modules/preact/src/jsx';
+import TargetedEvent = JSXInternal.TargetedEvent;
+import TargetedKeyboardEvent = JSXInternal.TargetedKeyboardEvent;
```

Comment:
We should probably use FormEvent, ChangeEvent, MouseEvent, KeyboardEvent, etc instead of importing from JSXInternal

Reply:
I fully agree, but TS complains when using them. Please see my above comments for more details